name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/DividendListWidget.java

```diff
@@ -0,0 +1,354 @@
+    {
+        public DateTypeConfig(WidgetDelegate<?> delegate)
+        {
+            super(delegate, Messages.LabelEarningsDividendDateType, DateType.class, Dashboard.Config.EARNING_TYPE,
+                            Policy.EXACTLY_ONE);
+        }
+    }
+
+    /**
+     * @deprecated This constructor is only used for testing, don't use it for
+     *             anything else
+     */
+    @Deprecated
+    DividendListWidget() // needed for testing
+    {
+        super(null, null);
+    }
+
+    public DividendListWidget(Widget widget, DashboardData data)
+    {
+        super(widget, data);
+
+        addConfig(new DateTypeConfig(this));
+        addConfig(new DateStartRangeConfig(this));
+        addConfig(new DateEndRangeConfig(this));
+        addConfig(new ChartHeightConfig(this));
+    }
+
+    @Override
+    public Supplier<List<DividendItem>> getUpdateTask()
+    {
+        return getUpdateTask(LocalDate.now());
+    }
+
+    Supplier<List<DividendItem>> getUpdateTask(LocalDate now)
+    {
+        return () -> {
+
+            DateType dateType = getDateTypeValue();
+
+            List<DividendItem> items = new ArrayList<>();
+            LocalDate fromDate = getStartRangeValue().getDate(now);
+            LocalDate untilDate = getEndRangeValue().getDate(now);
+            for (Security security : getSecurities())
+            {
+                for (SecurityEvent se : security.getEvents())
+                {
+                    if (!(se instanceof DividendEvent de))
+                    {
+                        continue;
+                    }
+                    DividendItem added = checkAndAdd(items, security, de, dateType, DateType.EX_DIVIDEND_DATE, fromDate,
+                                    untilDate, null);
+                    checkAndAdd(items, security, de, dateType, DateType.PAYMENT_DATE, fromDate, untilDate, added);
+                }
+            }
+
+            Collections.sort(items, new DividendItemComparator());
+            LocalDate prevDate = null;
+            DividendItem prevItem = null;
+            for (DividendItem item : items)
+            {
+                prevItem = item;
+                LocalDate currDate = item.getFirstType().getDate(item.div);
+                if (prevDate == null || !currDate.isEqual(prevDate))
+                {
+                    item.hasNewDate = true;
+                    prevDate = currDate;
+                    continue;
+                }
+            }
+
+            return items;
+        };
+    }
+
+    static class DividendItemComparator implements Comparator<DividendItem>
+    {
+
+        @Override
+        public int compare(DividendItem di1, DividendItem di2)
+        {
+            DateType type1 = di1.getFirstType();
+            DateType type2 = di2.getFirstType();
+            int ret = type1.getDate(di1.div).compareTo(type2.getDate(di2.div));
+            if (ret != 0)
+            {
+                return ret; //
+            }
```

Comment:
I know but I've got burned so many times with if-statements without braces. Google "apple goto fail" and you'll see another spectacular example of things that got burnt because of this.